Configuration/PyReleaseValidation/python/relval_upgrade.py

```diff
@@ -105,12 +105,20 @@ def makeStepName(key,frag,step,suffix):
             if upgradeDatasetFromFragment[frag]=="TTbar_13" and '2018' in key:
                 workflows[numWF+upgradeSteps['ParkingBPH']['offset']] = [ upgradeDatasetFromFragment[frag], stepList['ParkingBPH']]
 
+            inclPremix = 'PU' in key
+            if inclPremix:
+                inclPremix = False
+                for y in ['2021', '2023', '2024', '2026']:
+                    if y in key:
+                        inclPremix = True
+                        continue
```

Comment:
This piece of code is only about adding entries to the upgrade matrix. Does it still make sense to try to limit the workflow generation for specific years? I'm just thinking if it would be easier to generate them for all (leading to duplication between `relval_upgrade.py` and `relval_premix.py` for 2017 and 2018, but maybe it doesn't really matter much?)

Reply:
@makortel I guess that this could be part of the general cleaning that @chayanit is planning for PyReleaseValidation

Reply:
Thanks, I'll leave this part as it is then.

Reply:
Just to give my two cents: I am happy to have this kind of workflow present in the upgrade matrix (even if a similar workflow is present elsewhere). The only exception is if the workflow is known not to run or work correctly.